src/main/java/com/iexec/worker/executor/TaskNotificationService.java

```diff
@@ -205,10 +204,15 @@ protected void onTaskNotification(TaskNotification notification) {
     }
 
     private boolean storeWorkerpoolAuthorizationFromExtraIfPresent(TaskNotificationExtra extra) {
-        if (extra != null && extra.getWorkerpoolAuthorization() != null){
-            return contributionService.putWorkerpoolAuthorization(extra.getWorkerpoolAuthorization());
+        boolean success = true;
+        if(extra != null && extra.getWorkerpoolAuthorization() != null){
+            success = contributionService.putWorkerpoolAuthorization(extra.getWorkerpoolAuthorization());
```

Comment:
Why not inject and use directly `WorkerpoolAuthorizationService` ?
The rebound through `ContributionService` does not really make sense today.

Reply:
I agree, updated: https://github.com/iExecBlockchainComputing/iexec-worker/pull/475/commits/f8417e7de213770b7eb0191c40007cca616126af :heavy_check_mark:

Reply:
Thanks for this update